services/navigation/builtin/builtin_test.go

```diff
@@ -197,13 +197,53 @@ func TestStartWaypoint(t *testing.T) {
 		test.That(t, err, test.ShouldBeNil)
 
 		ns.(*builtIn).mode = navigation.ModeManual
-		err = ns.SetMode(ctx, navigation.ModeWaypoint, nil)
+		cancelCtx, fn := context.WithTimeout(ctx, time.Millisecond*10)
+		defer fn()
+		err = ns.SetMode(cancelCtx, navigation.ModeWaypoint, nil)
 		test.That(t, err, test.ShouldBeNil)
+		time.Sleep(time.Millisecond * 10)
+		ns.(*builtIn).wholeServiceCancelFunc()
 		ns.(*builtIn).activeBackgroundWorkers.Wait()
 
 		currentInputsShouldEqual(ctx, t, kinematicBase, pt)
 	})
 
+	t.Run("Switch from waypoint to manual", func(t *testing.T) {
```

Comment:
this looks really similar to the above function, maybe we could collapse them into one to avoid adding reduntant tests

Reply:
I agree it is super similar. I took a thorough look through the tests in this file and I believe that this test is actually a duplicate of one which already exists.
```
t.Run("Calling SetMode cancels current and future MoveOnGlobe calls", func(t *testing.T) {
```
for this reason, i will just remove this sub-test.